src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/EnvelopeProcessor.java

```diff
@@ -53,34 +53,31 @@ public Envelope processEnvelope(byte[] metadataStream) throws IOException {
     }
 
     public void markAsUploaded(Envelope envelope, String containerName, String zipFileName) {
-        envelopeStateRepository.save(
-            EnvelopeStatusBuilder
-                .newEnvelopeStatus(containerName, zipFileName)
-                .withEnvelope(envelope)
-                .withStatus(DOC_UPLOADED)
-                .build()
-        );
+        EnvelopeState state = new EnvelopeState(containerName, zipFileName);
+
+        state.setEnvelope(envelope);
+        state.setStatus(DOC_UPLOADED);
+
+        envelopeStateRepository.save(state);
     }
 
     public void markAsUploadFailed(String reason, Envelope envelope, String container, String zipFileName) {
-        envelopeStateRepository.save(
-            EnvelopeStatusBuilder
-                .newEnvelopeStatus(container, zipFileName)
-                .withEnvelope(envelope)
-                .withStatus(DOC_UPLOAD_FAILURE)
-                .withReason(reason)
-                .build()
-        );
+        EnvelopeState state = new EnvelopeState(container, zipFileName);
+
+        state.setEnvelope(envelope);
+        state.setStatus(DOC_UPLOAD_FAILURE);
+        state.setReason(reason);
```

Comment:
perhaps all could be moved to constructor?

Reply:
I'll move status only as it is required field